ledger/src/blockstore.rs

```diff
@@ -1531,25 +1533,29 @@ impl Blockstore {
         if index0.is_none() {
             return Ok(None);
         }
-        let mut index0 = self.transaction_status_index_cf.get(0)?.unwrap();
+        let mut index0 = index0.unwrap();
         let mut index1 = self.transaction_status_index_cf.get(1)?.unwrap();
 
+        let mut w_active_transaction_status_index =
```

Comment:
This lock needs to be grabbed before the get on index 0 and index 1, so you ensure you're not reading a stale version of those

Reply:
so as things stand now, I think you would actually have to hold this lock until the batch write is committed because you always need to commit both `index0` and `index1` in order to flip the "frozen" flag, and one of those objects is always getting written to by the status writer. 

I think we can just commit this write here, can you think of any reasons it needs to be in the WriteBatch to be atomic with the other operatioin?

Reply:
Yes, this reason: if the "frozen" flag flips first and then another transaction-status write comes in before the delete happens, it will get added to the index that then gets immediately deleted.

So I think you're right that we need to grab the lock in `run_purge` and hold it until the write is completed.

Reply:
Actually, I think it just needs to get held until the `delete_range_cf` gets added to the WriteBatch, right?

Reply:
> Yes, this reason: if the "frozen" flag flips first and then another transaction-status write comes in before the delete happens, it will get added to the index that then gets immediately deleted.
> 
> So I think you're right that we need to grab the lock in `run_purge` and hold it until the write is completed.

If we were to commit the changes (don't add them to the write batch, commit immediately) to the transaction indexes in the `toggle function`, then after we exit that function and release the transaction index lock, all transaction status writes should now be to the other column right, so this shouldn't happen?

Reply:
I see, so either we need to hold the lock until the end where this write batch is committed